### PR TITLE
Intangible Assets: use new NoteService,NoteType

### PR DESCRIPTION
**Pull request by ch-tstrickland**
To save unnecessary lines of code, the NoteService is a new area for
 common code, previously in classes such as
 IntangibleAssetsNoteService(Impl). This commit uses the new NoteService
 and deletes the old code that the NoteService replaces.
Add Intangible Assets type to NoteType.
Create IntangibleAssetHandler, IntangibleAssetDateHandler classes.
Update tests, add new tests.

Fixes BI-3565

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/web/accounts/service/notehandler/smallfull/IntangibleAssetsHandler.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import uk.gov.companieshouse.web.accounts.service.notehandler.NoteResourceHandler;
+import uk.gov.companieshouse.web.accounts.service.smallfull.SmallFullService;
+
+@Component
+public class IntangibleAssetsHandler implements NoteResourceHandler<IntangibleApi> {
+
+    @Autowired
+    private SmallFullService smallFullService;
+
+    private static final UriTemplate EMPLOYEES_URI = new UriTemplate(
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/web/accounts/service/notehandler/smallfull/IntangibleAssetsHandler.java:22]: Should this be `INTANGIBLE_URI`?

File: src/main/java/uk/gov/companieshouse/web/accounts/service/notehandler/smallfull/IntangibleAssetsHandler.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        return EMPLOYEES_URI.expand(transactionId, companyAccountsId).toString();
+    }
+
+    @Override
+    public Executor<ApiResponse<IntangibleApi>> get(ApiClient apiClient, String uri) {
+        return apiClient.smallFull().intangible().get(uri);
+    }
+
+    @Override
+    public Executor<ApiResponse<Void>> update(ApiClient apiClient, String uri, IntangibleApi employeesApi) {
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/web/accounts/service/notehandler/smallfull/IntangibleAssetsHandler.java:37]: Should the api object be renamed?

File: src/main/java/uk/gov/companieshouse/web/accounts/transformer/smallfull/intangible/impl/IntangibleAssetsTransformerImpl.java [hunk 3]
```diff
@@ -7,19 +7,20 @@
...
 @Component
-public class IntangibleAssetsTransformerImpl implements IntangibleAssetsTransformer {
+public class IntangibleAssetsTransformerImpl implements NoteTransformer<IntangibleAssets, IntangibleApi> {
 
     @Autowired
     private IntangibleAssetsTransformerFactory factory;
 
     @Override
-    public IntangibleAssets getIntangibleAssets(IntangibleApi intangibleApi) {
+    public IntangibleAssets toWeb(IntangibleApi intangibleApi) {
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/web/accounts/transformer/smallfull/intangible/impl/IntangibleAssetsTransformerImpl.java:23]: We need an additional check in this method for the case where `intangibleApi == null`

File: src/main/java/uk/gov/companieshouse/web/accounts/service/notehandler/smallfull/IntangibleAssetsHandler.java (hunk 1 above)
Comment (amoriartyCH) [inline src/main/java/uk/gov/companieshouse/web/accounts/service/notehandler/smallfull/IntangibleAssetsHandler.java:22]: should this be intangible_uri

Comment (amoriartyCH) [reply src/main/java/uk/gov/companieshouse/web/accounts/service/notehandler/smallfull/IntangibleAssetsHandler.java:22]: Just saw Brad has commented, will resolve this comment as its a duplicate.

File: src/main/java/uk/gov/companieshouse/web/accounts/transformer/smallfull/intangible/impl/IntangibleAssetsTransformerImpl.java (hunk 3 above)
Comment (ch-tstrickland) [reply src/main/java/uk/gov/companieshouse/web/accounts/transformer/smallfull/intangible/impl/IntangibleAssetsTransformerImpl.java:23]: > We need an additional check in this method for the case where `intangibleApi == null`

... which would then return null? Or just empty?

Comment (BSaunders95) [reply src/main/java/uk/gov/companieshouse/web/accounts/transformer/smallfull/intangible/impl/IntangibleAssetsTransformerImpl.java:23]: If the api object is null, we need to return an 'empty' object to back the page - see other transformers for examples 👍

Comment (ch-tstrickland) [reply src/main/java/uk/gov/companieshouse/web/accounts/transformer/smallfull/intangible/impl/IntangibleAssetsTransformerImpl.java:23]: Fixed.

**Pull request merged**